lib/web/integrations_awsoidc.go

```diff
@@ -302,21 +312,33 @@ func extractAWSRegionsFromQuery(r *http.Request) ([]string, error) {
 	return ret, nil
 }
 
+// regionsForListingDeployedDatabaseService fetches relevant aws regions, parse the regions query param, and returns the overlap.
+// If no query params are present, relevant regions are returned.
+// ex: relevant = ["us-west-1"]; params = []; returns ["us-west-1"]
+// ex: relevant = []; params = ["us-west-1"]; returns []
+// ex: relevant = ["us-west-1"]; params = ["us-west-1"]; returns ["us-west-1"]
+// ex: relevant = ["us-west-1"]; params = ["us-west-2"]; returns []
 func regionsForListingDeployedDatabaseService(ctx context.Context, r *http.Request, authClient databaseGetter, discoveryConfigsClient discoveryConfigLister) ([]string, error) {
+	relevant, err := fetchRelevantAWSRegions(ctx, authClient, discoveryConfigsClient)
```

Comment:
Nit: Might want to add a comment here explaining what "relevant regions" are. I had to look up the function to understand that it gathers the regions from active database services and discovery configs.